python/ml4ir/data/tfrecord_reader.py

```diff
@@ -49,6 +49,22 @@ def make_parse_fn(feature_config: FeatureConfig, max_num_records: int = 25) -> t
 
     punctuation_regex = "|".join([re.escape(c) for c in list(string.punctuation)])
 
+    @tf.function
```

Comment:
you should annotate what the expected input and output shapes and types are.  And we should have a unit test verifying it. ;)